BrainPortal/app/models/userfile.rb

```diff
@@ -727,15 +728,30 @@ def initialize_from_hash(atts = {}) #:nodoc:
         cb_error "Invalid :if condition '#{method}' in model." unless method.respond_to?(:to_proc)
         @conditions << method.to_proc
       end
+      @errors = []
     end
 
     def valid_for?(userfile) #:nodoc:
       return true if @conditions.empty?
-      @conditions.all? { |condition| condition.call(userfile) }
+
+      @conditions.all? do |condition|
+        [condition.call(userfile)].flatten.all? { |error| error == true}
```

Comment:
I don't get it. This method `valid_for?` returns true of ALL conditions evaluations return true errors?!?